web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,183 @@
+            // Core bus keypress doesn't seem to send arrow keys
+            core.bus.on('keydown', this, this.handleNavKeys);
+        },
+        
+        // It provides handlers to hide drawer when "unfocused"
+        handleClickZones: function() {
+            this.$el.drawer('close');
+            $('.oe_secondary_menus_container')
+                .parent()
+                .collapse('hide');
+        },
+        
+        // It provides keyboard shortcuts for app drawer nav
+        handleNavKeys: function(e) {
+            if (!this.isOpen){
+                return;
+            }
+            if (Object.keys(this.directionCodes).indexOf(e.keyCode.toString()) !== -1) {
+                var $link = this.findAdjacentAppLink(
+                    this.$selectedAppLink,
+                    this.directionCodes[e.keyCode][0],
+                    this.directionCodes[e.keyCode][1]
+                );
+                this.selectAppLink($link);
+            }
+        },
+        
+        onDrawerClose: function() {
+            this.isOpen = false;
+            this.$selectedAppLink = $(false);
+            core.bus.trigger('drawer.closed');
+            this.$el.one('drawer.opened', $.proxy(this.onDrawerOpen, this));
+        },
+        
+        // Generate x and y coords for app icons
+        onDrawerOpen: function() {
+            var self = this;
+            this.isOpen = true;
+            this.$appLinks = $('.css-app-drawer-icon-app').parent();
+            this.selectAppLink($(this.$appLinks[0]));
+            this.linkMap = {};  // Key by row, col
+            var x = 0,
+                y = -1,
+                lastTop = -9999,
+                newTop = null,
+                $lastVal = null;
+            _.each(this.$appLinks, function(val) {
+                $lastVal = $(val);
+                newTop = $lastVal.offset().top;
+                if (lastTop < newTop) {
+                    y ++;
+                    x = 0;
+                    self.linkMap[y] = {};
+                }
+                self.linkMap[y][x] = $lastVal;
+                $lastVal.data('pos-x', x).data('pos-y', y);
+                x ++;
+                lastTop = newTop;
+            });
+            this.$appLinks.on('mouseenter', $.proxy(this.selectAppLinkEvent, this));
+            this.$el.one('drawer.closed', $.proxy(this.onDrawerClose, this));
+            core.bus.trigger('drawer.opened');
+        },
+        
+        // It provides a proxy method allowing for app link select w/ events
+        selectAppLinkEvent: function(event) {
+            this.selectAppLink($(event.target));
+        },
+        
+        // It selects an app link visibly
+        selectAppLink: function($appLink) {
+            if ($appLink) {
+                this.$selectedAppLink = $appLink;
+                this.$appLinks.removeClass('keyboard-selected');
+                $appLink.addClass('keyboard-selected').focus();
```

Comment:
Oh sweet I didn't know there was already a class for the focus coloring. I still need to do the `.focus()` though because that is how the `Enter` key selects the href